src/trix/core/helpers/config.coffee

```diff
@@ -0,0 +1,21 @@
+Trix.extend
+
+  allAttributeNames: null
+  listAttributeNames: null
+
+  getAllAttributeNames: ->
+    @allAttributeNames ?= (
```

Comment:
The memoization here won’t behave the way you expect.

When you pull out a reference to `getAllAttributeNames` or `getListAttributeNames` and invoke it like this:

```coffee
{getAllAttributeNames} = Trix

for name in getAllAttributeNames()
  # ...
```

…the value of `this` in the `getAllAttributeNames` body will be `window`. That’s because the function isn’t explicitly bound to the `Trix` object and isn’t being called with a receiver.

What that means is the function will set an `allAttributeNames` property on the `window` object, polluting the global namespace.

If this were a class definition, we could use the fat arrow `=>` to define the function bound to the value of `this` we expect. In this case, we’re just defining an object literal, so that wouldn’t help.

Another solution would be to just replace `@allAttributeNames` with the fully-qualified `Trix.allAttributeNames`, avoiding referring to `this` at all. That solves the global `window` pollution, but still pollutes the `Trix` object with private properties we’d rather not expose.

The best solution is to pull these properties out as local variables in the file’s closure:

```coffee
allAttributeNames = null
listAttributeNames = null

Trix.extend
  getAllAttributeNames: ->
    allAttributeNames ?= (...)

  getListAttributeNames: ->
    listAttributeNames ?= (...)
```

By closing over those references, we can access them regardless of the value of `this`, and we don’t need to pollute the `Trix` object with extra private properties.